src/transactions/ParallelApplyUtils.cpp

```diff
@@ -682,7 +695,18 @@ ThreadParallelApplyLedgerState::getLiveEntryOpt(LedgerKey const& key) const
     {
         releaseAssert(!mThreadRestoredEntries.entryWasRestored(key));
     }
-    auto res = mLiveSnapshot->load(key);
+
+    // Check InMemorySorobanState cache for soroban types
+    std::shared_ptr<LedgerEntry const> res;
+    if (InMemorySorobanState::isInMemoryType(key))
+    {
+        res = mInMemorySorobanState.get(key);
```

Comment:
Is this thread safe? The comment at the top of `InMemorySorobanState` says it's not.

Reply:
I'll clarify the comment in `InMemorySorobanState`. It's not thread safe like the BucketList snapshot, where any thread can load at any time. It's "thread safe" where multiple threads can read concurrently so long as you join those threads before making an update, which is what we do in the apply path. I have a follow up PR that codifies this invariant within the LedgerManager